Lib/test/test_tools/test_msgfmt.py

```diff
@@ -240,6 +240,362 @@ def test_strings(self):
                     with self.assertRaises(Exception):
                         msgfmt.make('messages.po', 'messages.mo')
 
+    def test_general_syntax_errors(self):
+        invalid_po_files = (
+            '',
+            '"',
+            '""',
+            '"foo"',
+            # 'msgid',  # invalid but currently accepted
+            'msgstr',
+            'msgid_plural',
+            # 'msgctxt',  # invalid but currently accepted
+            'msgstr',
+            'msgstr[0]',
+            '[0]',
+
+            # unclosed string
+            '''
```

Comment:
You could use `textwrap.dedent()` and strip the initial newlines. It's up to you, but perhaps the indented code would look better.

Reply:
Added, I think it reads better with `textwrap.dedent()`